lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskRuntime.cs

```diff
@@ -174,6 +174,21 @@ public void Close(byte[] message)
                 _currentStatus.SetException(TaskClientCodeException.Create(
                     TaskId, ContextId, "Error during Close().", e));
             }
+            finally
+            {
+                try
+                {
+                    if (_userTask != null)
+                    {
+                        _userTask.Dispose();
+                    }
+                }
+                catch (Exception e)
+                {
+                    Utilities.Diagnostics.Exceptions.Caught(
```

Comment:
We have caught all the exceptions during the task closing. This is the exception during Dispose in finally block. As the task has been closed, therefore we log the error and ignore the exception.
Same code for StartTaskOnNewThread above in the same class.
It is a good point to document it. I have updated ITask comments for this.